docs/src/containers.md

```diff
@@ -1,13 +1,30 @@
+```@meta
+DocTestSetup = quote
+    using JuMP
+end
+```
+
 Containers
```

Comment:
There is a pretty good example (which made me understand the details of the containers) from @juan-pablo-vielma talk in JuliaCon that goes like this: 

```Julia 
model = Model(with_optimizer(GLPK.Optimizer))
@variable(model, x[i=1:5, j=1:5])                        # Array      <=> @variable(m, x[i=1:5, j=1:5], container = Auto)
set_1 = Base.OneTo(5)
@variable(model, y[i=set_1, j=1:5])                      # Array
set_2 = 1:5
@variable(model, z1[i=1:5, j=set_2])                     # JuMPArray
@variable(model, z2[i=1:5, j=set_2], container = Array)  # Array
a = 1
@variable(model, zz1[i=1:5, j=a:5])                      # JuMPArray
@variable(model, zz2[i=1:5, j=a:5], container = Array)   # Array
set_3 = [:a, :b, :c]
@variable(model, w[i=set_2, j=set_3])                    # JuMPArray
@variable(model, t[i=set_2, j=1:i])                      # Dict
@variable(model, h[i = 1:5; j=isodd(i)])                 # Dict
```

This is a short explanation of how powerful the container definition in JuMP is.

Reply:
This information *should* be covered already in the docs or in #1597, but I agree that this is a nice compact example to show off the different container options. @odow, thoughts on how snippets like this can be integrated into the docs?

Reply:
Moved this point to https://github.com/JuliaOpt/JuMP.jl/issues/1663.